tensorflow/stream_executor/rocm/rocm_blas.cc

```diff
@@ -63,161 +65,213 @@ namespace wrap {
-                                   __macro(rocblas_dtrmm)                      \
-                                   __macro(rocblas_ctrmm)                      \
-                                   __macro(rocblas_ztrmm) */                   \
-      __macro(rocblas_sgeam)                                                   \
-          __macro(rocblas_dgeam) /*  __macro(rocblas_cgeam)                    \
-                                   __macro(rocblas_zgeam)                      \
-                                   __macro(rocblas_sdgmm)                      \
-                                   __macro(rocblas_ddgmm)                      \
-                                   __macro(rocblas_cdgmm)                      \
-                                   __macro(rocblas_zdgmm) */
+#else
+
+#define STREAM_EXECUTOR_ROCBLAS_WRAP(__name)                              \
+  struct DynLoadShim__##__name {                                          \
+    static const char* kName;                                             \
+    using FuncPtrT = std::add_pointer<decltype(::__name)>::type;          \
+    static void* GetDsoHandle() {                                         \
+      auto s = internal::CachedDsoLoader::GetRocblasDsoHandle();          \
+      return s.ValueOrDie();                                              \
+    }                                                                     \
+    static FuncPtrT LoadOrDie() {                                         \
+      void* f;                                                            \
+      auto s = port::Env::Default()->GetSymbolFromLibrary(GetDsoHandle(), \
+                                                          kName, &f);     \
+      CHECK(s.ok()) << "could not find " << kName                         \
+                    << " in rocblas DSO; dlerror: " << s.error_message(); \
+      return reinterpret_cast<FuncPtrT>(f);                               \
+    }                                                                     \
+    static FuncPtrT DynLoad() {                                           \
+      static FuncPtrT f = LoadOrDie();                                    \
+      return f;                                                           \
+    }                                                                     \
+    template <typename... Args>                                           \
+    rocblas_status operator()(GpuExecutor* parent, Args... args) {        \
+      gpu::ScopedActivateExecutorContext sac{parent};                     \
+      return DynLoad()(args...);                                          \
+    }                                                                     \
+  } __name;                                                               \
+  const char* DynLoadShim__##__name::kName = #__name;
+
+#define STREAM_EXECUTOR_ROCBLAS_V2_WRAP(__name) \
+  STREAM_EXECUTOR_ROCBLAS_WRAP(__name)
+
+#endif
+
```

Comment:
add the // clang-format off pragma here so that the next `git clang-format ...` run on this file does not revert all your hard work :)